#!/usr/bin/env python
PACKAGE = "delta_manipulator"
 
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#REDS parameters
# gen.add("sp", double_t, 0, "delta measurement sp in m",    0.09192, 0.,   0.5)
# gen.add("sb", double_t, 0, "delta measurement sb in m",    0.57158, 0.,   0.6)
# gen.add("l", double_t, 0, "Distal joint length in m",    0.54535, 0.,   0.6)
# gen.add("L", double_t, 0, "Proximal joint length in m",    0.42545, 0.,   0.5)

#ARL parameters
# gen.add("sp", double_t, 0, "delta measurement sp in m",    0.04, 0.,   0.5)
# gen.add("sb", double_t, 0, "delta measurement sb in m",    0.16, 0.,   0.5)
# gen.add("l", double_t, 0, "Distal joint length in m",    0.15, 0.,   0.5)
# gen.add("L", double_t, 0, "Proximal joint length in m",    0.095, 0.,   0.5)

#Lachie parameters
gen.add("sp", double_t, 0, "delta measurement sp in m",    0.06, 0.,   0.5)
gen.add("sb", double_t, 0, "delta measurement sb in m",    0.34, 0.,   0.6)
gen.add("l", double_t, 0, "Distal joint length in m",    0.47, 0.,   0.6)
gen.add("L", double_t, 0, "Proximal joint length in m",    0.2, 0.,   0.5)

gen.add("dir1", bool_t, 0, "servo direction", True)
gen.add("dir2", bool_t, 0, "servo direction", True)
gen.add("dir3", bool_t, 0, "servo direction", True)

exit(gen.generate(PACKAGE, "delta_manipulator", "Delta"))